egps/egp-24/egp-24.ts

```diff
@@ -0,0 +1,109 @@
+import { ethers } from 'hardhat'
+
+// Artifacts
+import timelockData from '../../council/artifacts/contracts/features/Timelock.sol/Timelock.json'
+import treasuryData from '../../council/artifacts/contracts/features/Treasury.sol/Treasury.json'
+import coreVotingData from '../../council/artifacts/contracts/CoreVoting.sol/CoreVoting.json'
+
+// Helpers
+import * as addresses from '../helpers/addresses'
+import { createCallHash } from '../helpers/hashing'
+import { getExpiry } from '../helpers/expiry'
+import { AirdropReclaim__factory } from '../../typechain/factories'
+
+async function proposal() {
+  // Setup your signer
+  const [deployer] = await ethers.getSigners()
+  console.log(`current signer is ${deployer.address}`)
+
+  // Deploy the AirdropReclaim contract
+  console.log('deploying AirdropReclaim contract...')
+  const factory = new AirdropReclaim__factory(deployer)
+  let airdropReclaim = await factory.deploy()
+  airdropReclaim = await airdropReclaim.deployed()
+  console.log(`AirdropReclaim contract deployed to ${airdropReclaim.address}`)
+
+  // Setup your interfaces
+  const timelockInterface = new ethers.utils.Interface(timelockData.abi) // TimeLock is like sudo, you always need it.
+  const elfi = await ethers.getContractAt('ELFI', addresses.ELFI)
+
+  // Connect the signer to the coreVotingContract, this is where your proposals will fed into.
+  const coreVoting = new ethers.Contract(addresses.CoreVoting, coreVotingData.abi, deployer)
+
+  // --- main egp logic ---
+  /**
+   * Steps:
+   * 1. Transfer ownership of ELFI contract to AirdropReclaim contract
+   * 2. Call reclaim method on AirdropReclaim contract
+   * 3. Reclaim method on AirdropReclaim contract transfers ownership back to Timelock
+   */
+
+  const transferOwnershipCalldata = elfi.interface.encodeFunctionData('setOwner', [airdropReclaim.address])
```

Comment:
do we need this? Shouldn't we only need to call reclaim?

Reply:
Only the owner can call `mint()` and `burn()` functions.  The 3 steps in the comment above look correct to me.